apps/etterlatte-brev-api/src/main/kotlin/no/nav/etterlatte/brev/oversendelsebrev/OversendelseBrevService.kt

```diff
@@ -0,0 +1,280 @@
+import no.nav.etterlatte.brev.adresse.AdresseService
+import no.nav.etterlatte.brev.adresse.AvsenderRequest
+import no.nav.etterlatte.brev.behandling.PersonerISak
+import no.nav.etterlatte.brev.db.BrevRepository
+import no.nav.etterlatte.brev.distribusjon.DistribusjonService
+import no.nav.etterlatte.brev.distribusjon.DistribusjonsTidspunktType
+import no.nav.etterlatte.brev.distribusjon.DistribusjonsType
+import no.nav.etterlatte.brev.hentinformasjon.BrevdataFacade
+import no.nav.etterlatte.brev.hentinformasjon.SakService
+import no.nav.etterlatte.brev.model.Brev
+import no.nav.etterlatte.brev.model.BrevDataFerdigstilling
+import no.nav.etterlatte.brev.model.BrevDataFerdigstillingRequest
+import no.nav.etterlatte.brev.model.BrevID
+import no.nav.etterlatte.brev.model.BrevInnhold
+import no.nav.etterlatte.brev.model.BrevProsessType
+import no.nav.etterlatte.brev.model.Mottaker
+import no.nav.etterlatte.brev.model.OpprettNyttBrev
+import no.nav.etterlatte.brev.model.Pdf
+import no.nav.etterlatte.brev.model.Slate
+import no.nav.etterlatte.brev.model.Status
+import no.nav.etterlatte.brev.toMottaker
+import no.nav.etterlatte.libs.common.behandling.Klage
+import no.nav.etterlatte.libs.common.behandling.KlageUtfallMedData
+import no.nav.etterlatte.libs.common.behandling.SakType
+import no.nav.etterlatte.libs.common.feilhaandtering.GenerellIkkeFunnetException
+import no.nav.etterlatte.libs.common.feilhaandtering.UgyldigForespoerselException
+import no.nav.etterlatte.libs.common.person.Folkeregisteridentifikator
+import no.nav.etterlatte.libs.common.person.Vergemaal
+import no.nav.etterlatte.libs.common.tidspunkt.Tidspunkt
+import no.nav.etterlatte.token.BrukerTokenInfo
+import org.slf4j.LoggerFactory
+import java.time.LocalDate
+import java.util.UUID
+
+interface OversendelseBrevService {
+    fun hentOversendelseBrev(behandlingId: UUID): Brev?
+
+    suspend fun opprettOversendelseBrev(
+        behandlingId: UUID,
+        sakId: Long,
+        brukerTokenInfo: BrukerTokenInfo,
+    ): Brev
+
+    fun distribuerOversendelseBrev(
+        brevId: Long,
+        sakId: Long,
+        brukerTokenInfo: BrukerTokenInfo,
+    ): Brev
+
+    fun ferdigstillOversendelseBrev(
+        brevId: Long,
+        sakId: Long,
+        brukerTokenInfo: BrukerTokenInfo,
+    ): Pdf
+}
+
+class OversendelseBrevServiceImpl(
+    private val brevRepository: BrevRepository,
+    private val journalfoerBrevService: JournalfoerBrevService,
+    private val distribusjonService: DistribusjonService,
+    private val pdfGenerator: PDFGenerator,
+    private val sakService: SakService,
+    private val adresseService: AdresseService,
+    private val brevdataFacade: BrevdataFacade,
+) : OversendelseBrevService {
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    override fun hentOversendelseBrev(behandlingId: UUID): Brev? {
+        return brevRepository.hentBrevForBehandling(behandlingId, Brevtype.OVERSENDELSE_KLAGE).firstOrNull()
```

Comment:
Sikker på at `first` alltid er riktig?

Reply:
`singleOrNull` er kanskje bedre, for å kreve opprydding